### DISCUSSION TITLE
Update a running nuxt project without downtime

### DISCUSSION THREAD

**Discussion by d4rkr3pt0r** [nuxt/nuxt#9603] [Questions (Nuxt 2)]
I'm running my `Nuxt.js` projects using `pm2`. Every time I make some changes in components or pages or etc, I use `build command` and then I remove process using `pm2 delete all` and upload new build(`.nuxt folder`). This process would down website for couple of minutes. So i'm looking for a way to update build without downtime. Can anyone help?

**Comment by kissu** [comment]
If you're developing locally, why won't you use `npm run dev` or `yarn dev`?
The `build` command is to build it for production and it will take time indeed.

**Reply by d4rkr3pt0r** [reply]
Of course I'm developing locally. I was talking about the process of uploading updated `.nuxt folder` (on a running website) which was generated using `npm run build`. How can I do it without having downtime?

**Reply by kissu** [reply]
`.nuxt` is used to speed up things with cache, the interesting one is, `dist`. And you should not have any downtime if you do it properly aka: have a running website, build the new one (while the old one is still running), when the current build is done, swap it instead of the old currently running one. Heroku does that, mainly a devops question.

**Reply by d4rkr3pt0r** [reply]
@kissu `dist` folder in for static hosting. I'm talking about SSR website.

**Reply by kissu** [reply]
`dist` is used for both `target: static` and `target: server`.
